test/java/seedu/addressbook/commands/FindEmailCommandTest.java

```diff
@@ -0,0 +1,63 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+public class FindEmailCommandTest {
+
+    private final AddressBook addressBook = new TypicalPersons().getTypicalAddressBook();
+    private final TypicalPersons td = new TypicalPersons();
+
+    @Test
+    public void execute() throws IllegalValueException {
+        //same word, same case: matched
+        assertFindEmailCommandBehavior(new String[]{"ab@gmail.com"}, Arrays.asList(td.amy));
+
+        //same word, different case: not matched
+        assertFindEmailCommandBehavior(new String[]{"aB@gmail.com"}, Collections.emptyList());
+
+        //partial word: not matched
+        assertFindEmailCommandBehavior(new String[]{"alice"}, Collections.emptyList());
+
+        //multiple words: matched
+        assertFindEmailCommandBehavior(new String[]{"ab@gmail.com", "bc@gmail.com", "cd@gmail.com"},
+                Arrays.asList(td.amy, td.bill, td.candy));
+
+        //repeated keywords: matched
+        assertFindEmailCommandBehavior(new String[]{"ab@gmail.com", "ab@gmail.com"}, Arrays.asList(td.amy));
+
+        //Keyword matching a word in address: not matched
+        assertFindEmailCommandBehavior(new String[]{"Clementi"}, Collections.emptyList());
```

Comment:
Consider separating out the different assertions into different tests so that you can more easily pinpoint an errors when testing.

Do also use a name that can better explain the test.